integrasjonspunkt/src/test/java/no/difi/meldingsutveksling/cucumber/FiksIoInSteps.java

```diff
@@ -0,0 +1,130 @@
+package no.difi.meldingsutveksling.cucumber;
+
+import cucumber.api.java.Before;
+import cucumber.api.java.en.And;
+import lombok.RequiredArgsConstructor;
+import no.difi.meldingsutveksling.ServiceIdentifier;
+import no.difi.meldingsutveksling.dokumentpakking.service.CmsUtil;
+import no.difi.meldingsutveksling.domain.sbdh.*;
+import no.difi.meldingsutveksling.ks.fiksio.FiksIoSubscriber;
+import no.difi.meldingsutveksling.nextmove.NextMoveRuntimeException;
+import no.difi.meldingsutveksling.pipes.Plumber;
+import no.difi.meldingsutveksling.pipes.PromiseMaker;
+import no.difi.move.common.cert.KeystoreHelper;
+import no.ks.fiks.io.client.FiksIOKlient;
+import no.ks.fiks.io.client.SvarSender;
+import no.ks.fiks.io.client.model.AmqpChannelFeedbackHandler;
+import no.ks.fiks.io.client.model.KontoId;
+import no.ks.fiks.io.client.model.MeldingId;
+import no.ks.fiks.io.client.model.MottattMelding;
+import no.ks.fiks.io.client.send.FiksIOSender;
+import org.apache.commons.io.IOUtils;
+import org.springframework.beans.factory.ObjectProvider;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.PipedInputStream;
+import java.time.Duration;
+import java.util.UUID;
+import java.util.function.BiConsumer;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@RequiredArgsConstructor
+public class FiksIoInSteps {
+
+    private final Holder<Message> messageInHolder;
+    private final FiksIOKlient fiksIOKlient;
+    private final AsicFactory asicFactory;
+    private final Plumber plumber;
+    private final PromiseMaker promiseMaker;
+    private final ObjectProvider<CmsUtil> cmsUtilProvider;
+    private final KeystoreHelper keystoreHelper;
+    private final FiksIoSubscriber fiksIoSubscriber;
+
+    @Before
+    public void before() {
+        messageInHolder.reset();
+
+    }
+
+    @And("^FIKS IO prepares a message with messageId \"([^\"]+)\"$")
+    public void prepareFiksIoMessage(String messageId) {
+        // only needed for generating manifest through asicFactory
+        StandardBusinessDocumentHeader sbdh = new StandardBusinessDocumentHeader();
+        sbdh.addSender(new Sender().setIdentifier(new PartnerIdentification().setValue("0192:123123123")));
+        sbdh.addReceiver(new Receiver().setIdentifier(new PartnerIdentification().setValue("0192:321321321")));
+        StandardBusinessDocument sbd = new StandardBusinessDocument();
+        sbd.setStandardBusinessDocumentHeader(sbdh);
```

Comment:
Kan setjast i setup om den kun er for klargjerng og ikkje vert brukt?

Reply:
Dette er en "setup" per se. Den tar også parameter med verdier som trengs settes i sbd'en.